### PR TITLE
zoom-us: use flathub desktop integration 

### LINKED ISSUE #65678: zoom-us: Missing desktop icon
**Describe the bug**

`zoom-us` is missing its icon, because it does not appear in the source tarball.

I see a couple of options to fix:

1. Fetch the icon separately, perhaps from https://github.com/flathub/us.zoom.Zoom
2. Change the source to one of the packaged releases for another distro that includes the icon

cc @danbst 

Maintainer information:
```yaml
# a list of nixpkgs attributes affected by the problem
attribute: zoom-us
# a list of nixos modules affected by the problem
module:
```

### PR DESCRIPTION
**Pull request by worldofpeace**
###### Motivation for this change
Fixes #65678

###### Things done
Successfully launched `zoom-us` with the new desktop file.

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nix-review --run "nix-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/applications/networking/instant-messengers/zoom-us/default.nix [hunk 1]
```diff
@@ -66,15 +74,27 @@ in mkDerivation {
...
-    categories = "Network;Application;";
-    mimeType = "x-scheme-handler/zoommtg;";
-  }).buildCommand + ''
+  postInstall = ''
+    mkdir -p $out/share/{applications,appdata,icons}
+
+    # Desktop File
+    cp ${desktopIntegration}/us.zoom.Zoom.desktop $out/share/applications
+    substituteInPlace $out/share/applications/us.zoom.Zoom.desktop \
+        --replace "Exec=zoom" "Exec=$out/bin/zoom-us"
```
Comment (matthewbauer) [inline pkgs/applications/networking/instant-messengers/zoom-us/default.nix]: Note that this isn't strictly necessary. Although it also doesn't hurt. Eventually I would like an official policy on this, but right now we mix absolute Exec and PATH-based Exec's.

Comment (worldofpeace) [reply pkgs/applications/networking/instant-messengers/zoom-us/default.nix]: We definitely need some docs on guidelines for freedesktop things, and I think that would be one of them.
PATH-based Exec's have been a problem in the past when it needs some executable that
would be in `glib.dev` for example.

Comment (tadfisher) [reply pkgs/applications/networking/instant-messengers/zoom-us/default.nix]: Absolute Exec entries can be a pain when updating, though, as desktop files in the user profile aren't picked up. Is there a way to force, e.g., GNOME 3 to rebuild its desktop file database?

File: pkgs/applications/networking/instant-messengers/zoom-us/default.nix [hunk 2]
```diff
@@ -66,15 +74,27 @@ in mkDerivation {
...
+
+    # Appdata
+    cp ${desktopIntegration}/us.zoom.Zoom.appdata.xml $out/share/appdata
+
+    # Icons
+    icon_sizes=("64" "96" "128" "256")
+
+    for icon in ''${icon_sizes[*]}
```
Comment (tadfisher) [inline pkgs/applications/networking/instant-messengers/zoom-us/default.nix]: How about 

```bash
for size in 64 96 128 256; do
  ...
done
```

File: pkgs/applications/networking/instant-messengers/zoom-us/default.nix [hunk 3]
```diff
@@ -66,15 +74,27 @@ in mkDerivation {
...
+    do
+        mkdir -p $out/share/icons/hicolor/$icon\x$icon/apps
```
Comment (tadfisher) [inline pkgs/applications/networking/instant-messengers/zoom-us/default.nix]: In general, should we use two-space indents in scripts, to match the Nix convention?

File: pkgs/applications/networking/instant-messengers/zoom-us/default.nix [hunk 4]
```diff
@@ -22,12 +22,20 @@ let
...
+    sha256 = "0rm188844a10v8d6zgl2pnwsliwknawj09b02iabrvjw5w1lp6wl";
+  };
+
 in mkDerivation {
   name = "zoom-us-${version}";
 
   src = srcs.${stdenv.hostPlatform.system};
 
-  nativeBuildInputs = [ autoPatchelfHook makeWrapper wrapQtAppsHook ];
+  nativeBuildInputs = [ autoPatchelfHook ];
```
Comment (tadfisher) [inline pkgs/applications/networking/instant-messengers/zoom-us/default.nix]: Thanks for this, I didn't realize `wrapQtAppsHook` is pulled in by `qt5.mkDerivation`.

File: pkgs/applications/networking/instant-messengers/zoom-us/default.nix (hunk 1 above)
Comment (worldofpeace) [reply pkgs/applications/networking/instant-messengers/zoom-us/default.nix]: When you say
>  as desktop files in the user profile aren't picked up

Do you mean programs installed with `nix-env` don't get included or the entry in an application menu doesn't get updated?

Think that's a common issue #12757.

File: pkgs/applications/networking/instant-messengers/zoom-us/default.nix (hunk 3 above)
Comment (worldofpeace) [reply pkgs/applications/networking/instant-messengers/zoom-us/default.nix]: I don't think so. IIRC this is a rampant problem.

File: pkgs/applications/networking/instant-messengers/zoom-us/default.nix (hunk 2 above)
Comment (worldofpeace) [reply pkgs/applications/networking/instant-messengers/zoom-us/default.nix]: I think you'll find https://github.com/NixOS/nixpkgs/pull/65704/commits/b46a365a710fd9b1cb8ca95992176e87728d1fbc more readable.

File: pkgs/applications/networking/instant-messengers/zoom-us/default.nix (hunk 1 above)
Comment (tadfisher) [reply pkgs/applications/networking/instant-messengers/zoom-us/default.nix]: > the entry in an application menu doesn't get updated?

This one, and thanks. I don't think we need to address this here, so what you have is fine.

**Review by tadfisher — APPROVED** [review]
:+1: Thank you!

**Pull request merged**